lib/spack/spack/cmd/logs.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import shutil
+import sys
+import tempfile
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.util.compression as compression
+from spack.cmd.common import arguments
+
+description = "print out logs for packages"
+section = "basic"
+level = "long"
+
+
+def setup_parser(subparser):
+    arguments.add_common_arguments(subparser, ["spec"])
+
+
+def _dump_byte_stream_to_stdout(instream):
+    def write_as_is(byte_str, stream):
+        stream.write(byte_str)
+
+    def write_decoded(byte_str, stream):
+        stream.write(byte_str.decode("utf-8"))
+
+    write_to_stream = write_as_is
+
+    if sys.platform == "win32":
+        try:
+            outstream = sys.stdout.buffer
+        except AttributeError:
+            # e.g. a StringIO for unit tests
+            outstream = sys.stdout
+            write_to_stream = write_decoded
```

Comment:
I might be able to disable capsys to avoid this in tests. I'm marking this WIP in the meantime. Because this is specifically for dumping content to a terminal, I think we can choose not to support execution of this command in contexts where `sys.stdout` would not be a byte stream.

Reply:
These are now split up, and I added docs about the expectations for `stdout` when using this command.